### PR TITLE
feat: added getAllSchemaByDid method

### PR DESCRIPTION
**Pull request by tipusinghaw**
### What ###
- Added `getAllSchemaByDid` method
- Removed did resolver direct implementation

### CODE REVIEW

File: src/schema-manager.ts [hunk 1]
```diff
@@ -210,6 +215,37 @@ export class PolygonSchema {
...
+      const linkedResourceMetadata =
+        didDetails?.didDocumentMetadata?.linkedResourceMetadata
+      const schemaList: ResourcePayload[] = linkedResourceMetadata.filter(
+        (element: ResourcePayload) => {
+          if (element.resourceType === 'W3C-schema') {
+            return true
+          }
+          return false
+        },
+      )
```
Comment (sairanjit) [inline src/schema-manager.ts:240]:
```suggestion
      const schemaList: ResourcePayload[] = linkedResourceMetadata.filter(
        (element: ResourcePayload) => (element.resourceType === 'W3C-schema')
      )
```

**Review by sairanjit — CHANGES_REQUESTED** [review]

File: src/schema-manager.ts (hunk 1 above)
Comment (tipusinghaw) [reply src/schema-manager.ts]: Thanks for the suggestion, I have changed it

File: src/schema-manager.ts [hunk 2]
```diff
@@ -210,6 +215,31 @@ export class PolygonSchema {
...
+      if (!isValidDid) {
+        throw new Error('invalid did provided')
+      }
+      const didDetails = await this.resolver.resolve(did)
+
+      if (!didDetails?.didDocumentMetadata?.linkedResourceMetadata) {
+        throw new Error(
+          `The DID linked resource metadata for the given DID was not found!`,
+        )
+      }
```
Comment (sairanjit) [inline src/schema-manager.ts:230]: - I think by default if there are no schemas attached to the did instead of throwing error we should return empty array

**Review by sairanjit — CHANGES_REQUESTED** [review]

**Review by sairanjit — APPROVED** [review]
LGTM

**Pull request merged**